.github/workflows/protected-files-validation.yml

```diff
@@ -0,0 +1,81 @@
+name: 'Check Protected Files for Changes'
+
+on:
+  pull_request:
+    paths:
+      - eng/common/**/*
+      - eng/Version.Details.xml
+      - eng/Versions.props
+
+permissions:
+  pull-requests: read
+
+jobs:
+  check-version-details:
+    runs-on: ubuntu-22.04
+    if: ${{ !startsWith(github.event.pull_request.title, 'rebootstrap') && !startsWith(github.event.pull_request.title, 're-bootstrap') && !startsWith(github.event.pull_request.title, 'release') && !startsWith(github.event.pull_request.title, '.NET Source-Build ') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fetch base branch from upstream
+        run: |
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+
+      - name: Check for Version.Details.xml changes
+        working-directory: ${{ needs.setup.outputs.repo-path }}
+        run: |
+          if git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep '^eng/Version.Details.xml$' >/dev/null; then
+            echo "Version.Details.xml has changed."
+            exit 1
+          else
+            echo "No changes in Version.Details.xml."
+          fi
+
+  check-versions-props:
+    runs-on: ubuntu-22.04
+    if: ${{ !startsWith(github.event.pull_request.title, 'rebootstrap') && !startsWith(github.event.pull_request.title, 're-bootstrap') && !startsWith(github.event.pull_request.title, 'release') && !startsWith(github.event.pull_request.title, '.NET Source-Build ') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fetch base branch from upstream
+        run: |
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+
+      - name: Check for Versions.props changes
+        working-directory: ${{ needs.setup.outputs.repo-path }}
+        run: |
+          if git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep '^eng/Versions.props$' >/dev/null; then
```

Comment:
There are scenarios in which the Versions.props file can have valid edits outside of a rebootstrap/release PR.  e.g. you are working on the tasks and need to add a new dependency to the system. What is the expected workflow for these scenarios?  Is merge on red the only option?  Would it make sense to allow a "special" comment in the PR to indicate the edit is allowed by this check?

Reply:
IMO merge on red is sufficient.

Reply:
I am fine with merge on red as well.  I wanted to raise this and to ensure an explicit decision was made.

Reply:
> I am fine with merge on red as well. I wanted to raise this to ensure an explicit decision was made.